@here/harp-materials/lib/ShaderChunks/ExtrusionChunks.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export default {
+    extrusion_pars_vertex: `
+uniform float extrusionRatio;
+`,
+    extrusion_vertex: `
+transformed.z = transformed.z * extrusionRatio;
```

Comment:
This will not work for the SphereProjection, instead you have to multiply the normal of the tile withe the height of the vertex and the extrusionRatio: 
```suggestion
float height = position.w; 
transformed = transformed + normal * height * extrusionRatio;
```

Reply:
Thank you for pointing out!

Reply:
Could that work if the position attribute always remains at 3 components?